FirebaseAuth/Tests/SampleSwift/AuthenticationExample/CustomViews/LoginView.swift

```diff
@@ -12,162 +12,179 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import UIKit
-
-/// Login View presented when performing Email & Password Login Flow
-class LoginView: UIView {
-  var emailTextField: UITextField! {
-    didSet {
-      emailTextField.textContentType = .emailAddress
+import SwiftUI
+
+import FirebaseAuth
+
+struct LoginView: View {
+  @Environment(\.dismiss) private var dismiss
+  @State private var multiFactorResolver: MultiFactorResolver? = nil
+  @State private var onetimePasscode = ""
+  @State private var showingAlert = false
+
+  @State private var email: String = ""
+  @State private var password: String = ""
+
+  var body: some View {
+    Group {
+      VStack {
+        Group {
+          HStack {
+            VStack {
+              Text("Email/Password Auth")
+                .font(.title)
+                .bold()
+            }
+            Spacer()
+          }
+          HStack {
+            Text(
+              "Login or create an account using the Email/Password auth " +
+                "provider.\n\nEnsure that the Email/Password provider is " +
+                "enabled on the Firebase console for the given project."
+            )
+            .fixedSize(horizontal: false, vertical: true)
+            Spacer()
+          }
+        }
+        .padding(.vertical)
+
+        Spacer()
+        TextField("Email", text: $email)
+          .textFieldStyle(SymbolTextField(symbolName: "person.crop.circle"))
+        TextField("Password", text: $password)
+          .textFieldStyle(SymbolTextField(symbolName: "lock.fill"))
+        Spacer()
+        Group {
+          LoginViewButton(
+            text: "Login",
+            accentColor: .white,
+            backgroundColor: .orange
+          ) {
+            Task {
```

Comment:
I'd refactor the code in this closure into a function on the view. Or maybe even into a view model.

Reply:
Refactored out into a function on the view. I'll leave this thread open to consider doing a view model. For now, I'll focus on refactoring the UI bits..